tests/Jellyfin.Common.Tests/Extensions/StringExtensionsTests.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using MediaBrowser.Common.Extensions;
+using Xunit;
+
+namespace Jellyfin.Common.Tests.Extensions
+{
+    public class StringExtensionsTests
+    {
+        [Theory]
+        [InlineData("Banana split", ' ', "Banana")]
+        [InlineData("Banana split", 'q', "Banana split")]
+        public void LeftPart_ValidArgsCharNeedle_Correct(string str, char needle, string result)
+        {
+            Assert.Equal(result, str.AsSpan().LeftPart(needle).ToString());
```

Comment:
I think it's nice to separate the test logic that is performed from the test assertions. This is so much easier to read at a glance IMO
```suggestion
            var result = str.AsSpan().LeftPart(needle).ToString();
            Assert.Equal(expectedResult, result);
```
(FYI this suggestion can't be applied directly, you'd need to rename the `result` parameter as well)

Reply:
Would it be possible to test that the result is pointing at the same point in memory as the input string. Since the method is using `Span`. (This is less of a suggestion and more of a question to see if I'm correctly understanding how this works  😅)

Reply:
Assert.Method is a pretty common method in my experience, I think it's fine to use them.

Reply:
> Would it be possible to test that the result is pointing at the same point in memory as the input string. Since the method is using Span. (This is less of a suggestion and more of a question to see if I'm correctly understanding how Span works sweat_smile)

We could, but that would require unsafe code